core/services/pipeline/common.go

```diff
@@ -28,7 +28,7 @@ type (
 	Task interface {
 		Type() TaskType
 		DotID() string
-		Run(ctx context.Context, taskRun TaskRun, inputs []Result) Result
+		Run(ctx context.Context, meta JSONSerializable, inputs []Result) Result
```

Comment:
the only thing used from the taskRun was the metadata, just making that explicit